File: 1443/1443.c
int minimumTimeToCollectAllApplesI(struct TreeNode* root) {
    if (!root) return 0;
    int l = minimumTimeToCollectAllApplesI(root->left);
    int r = minimumTimeToCollectAllApplesI(root->right);
    return 1 + (l > r ? l : r);
}

File: 1443/1443.js
/**
 * @param {TreeNode} root
 * @return {number}
 */
var minimumTimeToCollectAllApplesI = function(root) {
    if (!root) return 0;
    return 1 + Math.max(minimumTimeToCollectAllApplesI(root.left), minimumTimeToCollectAllApplesI(root.right));
};